firedrake/preconditioners/low_order.py

```diff
@@ -1,17 +1,38 @@
 from firedrake.preconditioners.pmg import PMGPC, PMGSNES
 
-__all__ = ("P1PC", "P1SNES")
+__all__ = ("P1PC", "P1SNES", "LORPC")
 
 
 class P1PC(PMGPC):
+    """A two-level preconditioner with agressive p-coarsening."""
     def coarsen_element(self, ele):
         if super().max_degree(ele) <= self.coarse_degree:
             raise ValueError
         return super().reconstruct_degree(ele, self.coarse_degree)
 
 
 class P1SNES(PMGSNES):
+    """A two-level nonlinear solver with agressive p-coarsening."""
     def coarsen_element(self, ele):
         if super().max_degree(ele) <= self.coarse_degree:
             raise ValueError
         return super().reconstruct_degree(ele, self.coarse_degree)
+
+
+class LORPC(PMGPC):
+    """A low-order refined preconditioner with a P1-iso-Pk coarse space."""
```

Comment:
Does this produce the correct sparsity in the low order operator or is that separate work to come?

Reply:
This will just keep the high-order sparsity pattern with zero entires. We plan to add in a ChoppedPC, that drops zero entries as a first step. Utimately we would like The Right Thing™, one candidate for this is to use facet integral kernels on codimension-0 facets (the subcells). This is very similar to the dS/ds kernels. We would need a subcell-to-node map, which shouldn't be hard to construct from the cell to node map.